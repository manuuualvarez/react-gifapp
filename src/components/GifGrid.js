import React from 'react'
// Components
import {  GifGridItem } from './GifGridItem'
// Hooks
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ( {category }) => {

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> { category } </h3>
        
            { loading && <p className="animate__animated animate__flash">Loading...</p>  }

            <div className="card-grid">
                {  images.map( img =>  

                    <GifGridItem  
                        key={ img.id }
                        { ...img }
                    />       
                    )
                }
            </div>
        </>
    )
}
