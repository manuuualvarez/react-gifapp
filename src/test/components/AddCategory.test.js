import React from 'react';
import { shallow } from 'enzyme'; 
import {  PropTypes } from 'prop-types';
import '@testing-library/jest-dom';

// Components
import { AddCategory } from '../../components/AddCategory';

describe('Test on <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    // Life cycle
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories= {setCategories}/>);
    });

    // Method Test
    test("Component show succesfully", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Input text box should change", () => {
        const input = wrapper.find('input');
        const value = 'Hello word!';
        input.simulate('change', { target: { value: value } });
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test("Submit form does not post info if the text lenght is minor than 2", () => {
        wrapper.find('form').simulate('submit', { 
            preventDefault(){} 
        });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test("Should call setCategories and clear the inbox text", () => {
        // Simulate input change
        const value = 'Hello word!';
        wrapper.find('input').simulate('change', { target: { value: value } });
        // Simulate submit form
        wrapper.find('form').simulate('submit', { 
            preventDefault(){} 
        });
        // Set new value
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        // Clear the inputValue
        expect(wrapper.find('input').prop('value')).toBe('');
        

    })
})