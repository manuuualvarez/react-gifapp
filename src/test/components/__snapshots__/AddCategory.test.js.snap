// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AddCategory /> Component show succesfully 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
